src/Core/BaseSettings.cpp

```diff
@@ -31,11 +32,19 @@ void BaseSettingsHelpers::writeFlags(Flags flags, WriteBuffer & out)
 }
 
 
-BaseSettingsHelpers::Flags BaseSettingsHelpers::readFlags(ReadBuffer & in)
+UInt64 BaseSettingsHelpers::readFlags(ReadBuffer & in)
 {
     UInt64 res;
     readVarUInt(res, in);
-    return static_cast<Flags>(res);
+    return res;
+}
+
+SettingsTierType BaseSettingsHelpers::getTier(UInt64 flags)
+{
+    int8_t tier = (flags & Flags::TIER);
```

Comment:
```suggestion
    uint8_t tier = (flags & Flags::TIER);
```

Reply:
`SettingsTierType` is a int8_t for compatibility with `DataTypeEnum8`. If we use `uint8_t` here then we need to cast it again to use it. I'll add an explicit `static_cast<int8_t>` here for clarity, but I expect it will be the same the compiler is doing anyway (do the AND, get a UInt64, cast it to `int8_t`).